tests/convert-to-ts/C7203002-to-ts.test.ts

```diff
@@ -18,327 +12,176 @@ it("should convert C7203002 into typescript file", async () => {
-<silkscreenpath route={[{"x":4.101090350000163,"y":-10.49997899999994},{"x":4.578838950000204,"y":-10.49997899999994}]} />
-<silkscreenpath route={[{"x":1.561090350000086,"y":-10.49997899999994},{"x":2.0388389500001267,"y":-10.49997899999994}]} />
-<silkscreenpath route={[{"x":-0.9789096499999914,"y":-10.49997899999994},{"x":-0.5011610499999506,"y":-10.49997899999994}]} />
-<silkscreenpath route={[{"x":-3.5189096499998413,"y":-10.49997899999994},{"x":-3.0411610499998005,"y":-10.49997899999994}]} />
-<silkscreenpath route={[{"x":-6.058909649999919,"y":-10.49997899999994},{"x":-5.581161049999878,"y":-10.49997899999994}]} />
-<silkscreenpath route={[{"x":-8.598909649999996,"y":-10.49997899999994},{"x":-8.121161049999955,"y":-10.49997899999994}]} />
-<silkscreenpath route={[{"x":-11.138909649999846,"y":-10.49997899999994},{"x":-10.661161049999805,"y":-10.49997899999994}]} />
-<silkscreenpath route={[{"x":-13.678909649999923,"y":-10.49997899999994},{"x":-13.201161049999882,"y":-10.49997899999994}]} />
-<silkscreenpath route={[{"x":-16.218909649999887,"y":-10.49997899999994},{"x":-15.74116104999996,"y":-10.49997899999994}]} />
-<silkscreenpath route={[{"x":-18.75890964999985,"y":-10.49997899999994},{"x":-18.28116104999981,"y":-10.49997899999994}]} />
-<silkscreenpath route={[{"x":-21.298909649999928,"y":-10.49997899999994},{"x":-20.821161049999887,"y":-10.49997899999994}]} />
-<silkscreenpath route={[{"x":-23.838909649999778,"y":-10.49997899999994},{"x":-23.361161049999964,"y":-10.49997899999994}]} />
-      </footprint>}
-    />
-  )
-}
+    const pinLabels = {
+      pin1: ["GP0"],
+      pin2: ["GP1"],
+      pin3: ["GND8"],
+      pin4: ["GP2"],
+      pin5: ["GP3"],
+      pin6: ["GP4"],
+      pin7: ["GP5"],
+      pin8: ["GND6"],
+      pin9: ["GP6"],
+      pin10: ["GP7"],
+      pin11: ["GP8"],
+      pin12: ["GP9"],
+      pin13: ["GND4"],
+      pin14: ["GP10"],
+      pin15: ["GP11"],
+      pin16: ["GP12"],
+      pin17: ["GP13"],
+      pin18: ["GND2"],
+      pin19: ["GP14"],
+      pin20: ["GP15"],
+      pin21: ["GP16"],
+      pin22: ["GP17"],
+      pin23: ["GND1"],
+      pin24: ["GP18"],
+      pin25: ["GP19"],
+      pin26: ["GP20"],
+      pin27: ["GP21"],
+      pin28: ["GND3"],
+      pin29: ["GP22"],
+      pin30: ["RUN"],
+      pin31: ["GP26"],
+      pin32: ["GP27"],
+      pin33: ["GND5"],
+      pin34: ["GP28"],
+      pin35: ["ADC_VREF"],
+      pin36: ["pin36"],
+      pin37: ["3V3_EN"],
+      pin38: ["GND7"],
+      pin39: ["VSYS"],
+      pin40: ["VBUS"],
+      pin41: ["SWCLK"],
+      pin42: ["GND9"],
+      pin43: ["SWDIO"],
+      pin44: ["TP1"],
+      pin45: ["TP2"],
+      pin46: ["TP3"],
+      pin47: ["TP4"],
+      pin48: ["TP5"],
+      pin49: ["TP6"]
+    } as const
 
-export const usePICO = createUseComponent(PICO, pinLabels)"
-`)
+    export const PICO = (props: ChipProps<typeof pinLabels>) => {
+      return (
+        <chip
+          {...props}
```

Comment:
i just noticed, {...props} should be spread AFTER all the parameters, otherwise the user can't override properties, can you add? will tip